WordPress/Classes/Stores/StatsInsightsStore.swift

```diff
@@ -981,3 +985,66 @@ private extension InsightStoreState {
     }
 
 }
+
+// MARK: - iOS 14 Widgets Data
+private extension InsightStoreState {
+
+    private func saveHomeWidgetTodayData(stats: TodayWidgetStats) {
+
+        guard let siteID = SiteStatsInformation.sharedInstance.siteID else {
+            return
+        }
+
+        var homeWidgetTodayCache = HomeWidgetTodayData.read() ?? initializeHomeWidgetTodayData()
+
+        guard let oldData = homeWidgetTodayCache[siteID.intValue] else {
+            DDLogError("HomeWidgetToday: Failed to find a matching site")
+            return
+        }
+        // TODO - TODAYWIDGET: it might be better to move the blog updates in SiteStatsInformation
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        guard let blog = blogService.blog(byBlogId: siteID) else {
+            DDLogError("HomeWidgetToday: the site does not exist anymore")
+            // if for any reason that site does not exist anymore, remove it from the cache.
+            homeWidgetTodayCache.removeValue(forKey: siteID.intValue)
+            HomeWidgetTodayData.write(data: homeWidgetTodayCache)
+            return
+        }
+        // refresh stats and update any blog info, if they had changed
+        homeWidgetTodayCache[siteID.intValue] = HomeWidgetTodayData(siteID: siteID.intValue,
+                                                           siteName: blog.title ?? oldData.siteName,
+                                                           url: blog.url ?? oldData.url,
+                                                           timeZoneName: blogService.timeZone(for: blog).identifier,
+                                                           date: Date(),
+                                                           stats: stats)
+
+        HomeWidgetTodayData.write(data: homeWidgetTodayCache)
+        if #available(iOS 14.0, *) {
+            WidgetCenter.shared.reloadTimelines(ofKind: "WordPressHomeWidgetToday")
+        }
+    }
+
+    // TODO - TODAYWIDGET: this method will likely be exposed in the future to handle the creation of the file at login
+    private func initializeHomeWidgetTodayData() -> [Int: HomeWidgetTodayData] {
+
+        let blogService = BlogService(managedObjectContext: ContextManager.shared.mainContext)
+
+        return blogService.blogsForAllAccounts().reduce(into: [Int: HomeWidgetTodayData]()) {
+            if let blogID = $1.dotComID,
```

Comment:
As this isn't just a one-liner, it might be slightly clearer to name the variables here, instead of using $0 and $1

Reply:
+1

Reply:
I personally like the compactness of implicit arguments and find them quite readable, but since it seems that I am the minority 😄 I'm gonna change them to explicit